### ISSUE TITLE
Separate "test" and "validation" sets when dealing with large data 

### ISSUE THREAD

**Issue by earlbellinger** [astroswego/plotypus#41]
The user should be able to specify what fraction of the data should be left out for the final test set. For small data it can be 0, but for data like Kepler it should be around half. In the absense of user input, determine the fraction in some reasonable way by looking at the number of points available. 

The process will go like this: 
1. Parse the data 
2. Split the data in half (or whatever fraction) 
3. Determine the model by separating one of the halves into k training and validation sets and doing cross validation 
4. Finally, calculate the R^2 and MSE on the other half
